### PR TITLE
NH-88274 Add support for SW_APM_EXPORT_METRICS_ENABLED

### PR DESCRIPTION
**Pull request by tammy-baylis-swi**
Adds support for `SW_APM_EXPORT_METRICS_ENABLED=true/false` as an **opt-in** for OTLP metrics export. A documentation update PR will follow.

Under the hood, custom distro does several `setdefault` of configs. Please note that, even though `setdefault` is called for metrics endpoint, headers, etc, the metrics exporter will only be initialized by the Configurator if `SW_APM_EXPORT_METRICS_ENABLED=true`. Another key part is that `setdefault` only sets defaults; user can configure their own OTel metrics endpoint, headers, etc via environment variable and those will be respected by Configurator component inits.

To get all of metrics, logs and traces to export by OTLP, need to set all of `SW_APM_EXPORT_METRICS_ENABLED`, `SW_APM_EXPORT_LOGS_ENABLED`, and `OTEL_EXPORTER_OTLP_PROTOCOL` (one of `http/protobuf`, `grpc`).

See ticket comments for manual testing results.

Please let me know if any thoughts, questions, concerns! 😺

### CODE REVIEW

File: solarwinds_apm/configurator.py [hunk 1]
```diff
@@ -313,13 +313,15 @@ def _configure_traces_exporter(
...
+        """Configure traces exporters if agent enabled.
+        Links to global TracerProvider.
 
         Initialization of SolarWinds exporter requires a liboboe reporter
-        Note: if reporter is no-op, the SW exporter will not export spans."""
+        If `reporter` is no-op, the SW exporter will not export spans."""
         if not apm_config.agent_enabled:
-            logger.error("Tracing disabled. Cannot set trace exporter.")
+            logger.error(
+                "APM Python library disabled. Cannot set traces exporters."
```
Comment (tammy-baylis-swi) [inline solarwinds_apm/configurator.py:323]: Not changing `_configure_traces_exporter`, just updating docstring and error log for consistency.

File: tests/unit/test_configurator/test_configurator_metrics_exporter.py [hunk 2]
```diff
@@ -36,11 +36,30 @@ def test_configure_metrics_exporter_disabled(
...
+        test_configurator = configurator.SolarWindsConfigurator()
+        test_configurator._configure_metrics_exporter(
+            mock_apmconfig_metrics_enabled_false,
+        )
+        mock_pemreader.assert_not_called()
+        trace_mocks.get_tracer_provider.assert_not_called()
+        trace_mocks.get_tracer_provider().get_tracer.assert_not_called()
+        mock_meterprovider.assert_not_called()
 
-    def test_configure_metrics_exporter_flag_not_set(
```
Comment (tammy-baylis-swi) [inline tests/unit/test_configurator/test_configurator_metrics_exporter.py:40]: This must have tested an old flag that was temporarily implemented during early Lambda development. At some point I removed the flag but not the test, which didn't seem to be testing anything meaningful anymore. So I've repurposed to test new metrics enabled config.

File: solarwinds_apm/configurator.py [hunk 3]
```diff
@@ -384,8 +386,21 @@ def _configure_metrics_exporter(
...
-        """Configure SolarWinds OTel metrics exporters if any configured.
-        Links them to new metric readers and global MeterProvider."""
+        """Configure OTel OTLP metrics exporters if enabled and agent enabled.
+        Settings precedence: OTEL_* > SW_APM_EXPORT_METRICS_ENABLED.
+        Links to new metric readers and global MeterProvider."""
+        if not apm_config.agent_enabled:
+            logger.error(
+                "APM Python library disabled. Cannot set metrics exporters."
+            )
+            return
```
Comment (tammy-baylis-swi) [inline solarwinds_apm/configurator.py:396]: This was previously missing for `_configure_metrics_exporter`, so I've added it now.

File: solarwinds_apm/configurator.py [hunk 4]
```diff
@@ -422,11 +437,19 @@ def _configure_metrics_exporter(
...
+                # Inf interval to not invoke periodic collection
+                reader = PeriodicExportingMetricReader(
+                    exporter,
+                    export_interval_millis=math.inf,
+                )
+            else:
+                # Use default interval 60s else OTEL_METRIC_EXPORT_INTERVAL
+                reader = PeriodicExportingMetricReader(
+                    exporter,
+                )
```
Comment (tammy-baylis-swi) [inline solarwinds_apm/configurator.py:452]: This fixes export outside of lambda which wasn't tested much before.

File: solarwinds_apm/apm_config.py [hunk 5]
```diff
@@ -169,6 +170,9 @@ def __init__(
         self.metric_format = self._calculate_metric_format()
         self.certificates = self._calculate_certificates()
         self.__config["export_logs_enabled"] = self._calculate_logs_enabled()
+        self.__config["export_metrics_enabled"] = (
```
Comment (jerrytfleung) [inline solarwinds_apm/apm_config.py:173]: Do we need the bracket? I don't see the same in line 172.

**Review by jerrytfleung — APPROVED** [review]
LGTM. I left a comment about nits.

File: solarwinds_apm/apm_config.py (hunk 5 above)
Comment (tammy-baylis-swi) [reply solarwinds_apm/apm_config.py:173]: The linter did this, probably for slightly longer line char length.

**Pull request merged**